### DISCUSSION TITLE
How do I pad strings with custom fill charcter

### DISCUSSION THREAD

**Discussion by bytefoot** [Textualize/rich#1303] [Q&A]
I' have just discovered this awesome library, so i was just trying to incorporate this into on of my projects. I would like to know to how to justify strings with a particular fill/padding character other than the default space.

Currently, my code is like this:
```
from rich.console import Console
console = Console()
console.print("Hello World!", justify='center')
````
This works as expected and gives the result "Hello World!" padded with space. But how do i get it padded with some other character?

**Comment by willmcgugan** [comment]
There is a Padding class, but that only uses spaces. I didn't occur to me that anyone would want to pad with anything else.

However, a `rule` might be want you want. Try this...

```python
from rich.console import Console
console = Console()
console.rule("[b]Hello, World", characters="*-")
```

**Reply by bytefoot** [reply]
I guess this hack might just fill-in my requirement. Though the spaces between title and fill wasn't expected, this might suffice. I hope this padding feature would be considered in the future. Thanks.
